# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1234976036 18000
#      Wed Feb 18 11:53:56 2009 -0500
# Node ID 995f8b064b323fef2aac074ad162e5b05d1c1360
# Parent  35ea4f816bc95b9a7fa4c7dc1712165541f4e383
subplot.m: Compatible placement of subplots.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-02-18  Ben Abbott  <bpabbott@mac.com>
+
+	* plot/subplot.m: Fix unintended deletion of axes.
+
 2009-02-18  Frederick Umminger  <Frederick_Umminger@playstation.sony.com>
 
 	* audio/wavread.m: Improve search for data chunks.
 
 2009-02-18  Ivan Sutoris  <ivan.sutoris@gmail.com>
 
 	* miscellaneous/edit.m: Expand documentation about editor mode.
 
diff --git a/scripts/plot/subplot.m b/scripts/plot/subplot.m
--- a/scripts/plot/subplot.m
+++ b/scripts/plot/subplot.m
@@ -98,17 +98,17 @@ function h = subplot (rows, columns, ind
 
   if (columns < 1 || rows < 1 || index < 1)
     error ("subplot: columns,rows,index must be be positive");
   endif
 
   units = get (0, "defaultaxesunits");
   unwind_protect
     set (0, "defaultaxesunits", "normalized")
-    pos = subplot_position (rows, columns, index, "outerposition", units);
+    pos = subplot_position (rows, columns, index, "position", units);
 
     cf = gcf ();
 
     set (cf, "nextplot", "add");
 
     found = false;
     kids = get (cf, "children");
     for child = reshape (kids, 1, numel (kids))
@@ -120,17 +120,17 @@ function h = subplot (rows, columns, ind
         continue;
       endif
       if (strcmp (get (child, "type"), "axes"))
         ## Skip legend and colorbar objects.
         if (strcmp (get (child, "tag"), "legend") || 
 	    strcmp (get (child, "tag"), "colorbar"))
           continue;
         endif
-        objpos = get (child, "outerposition");
+        objpos = get (child, "position");
         if (all (objpos == pos))
 	  ## If the new axes are in exactly the same position as an
 	  ## existing axes object, use the existing axes.
 	  found = true;
 	  tmp = child;
         else
 	  ## If the new axes overlap an old axes object, delete the old
 	  ## axes.
@@ -147,16 +147,17 @@ function h = subplot (rows, columns, ind
 	  endif
         endif
       endif
     endfor
 
     if (found)
       set (cf, "currentaxes", tmp);
     else
+      pos = subplot_position (rows, columns, index, "outerposition", units);
       pos2 = subplot_position (rows, columns, index, "position", units);
       tmp = axes ("outerposition", pos, "position", pos2);
     endif
 
   unwind_protect_cleanup
     set (0, "defaultaxesunits", units);
   end_unwind_protect
 
@@ -193,17 +194,17 @@ function pos = subplot_position (rows, c
   width = width / columns;
   height = 1 - margins.top - margins.bottom - (rows-1)*margins.row;
   height = height / rows;
 
   if (strcmp (position_property, "outerposition") )
     ## Calculate the outerposition/position inset
     if (rows > 1)
       inset.top    = 8/420;
-      inset.bottom = max (polyval ([0.1382,-0.0026], width), 16/420);
+      inset.bottom = max (polyval ([0.1382,-0.0026], height), 16/420);
     else
       inset.bottom = margins.bottom;
       inset.top = margins.top;
     endif
     if (columns > 1)
       if (strcmpi (units, "normalized"))
         inset.right = max (polyval ([0.1200,-0.0014], width), 5/560);
       else
@@ -223,16 +224,21 @@ function pos = subplot_position (rows, c
 
   yp = fix ((index(:)-1)/columns);
   xp = index(:) - yp*columns - 1;
   yp = (rows - 1) - yp;
 
   x0 = xp .* (width + margins.column) + margins.left;
   y0 = yp .* (height + margins.row) + margins.bottom;
 
+  if (strcmp (position_property, "outerposition") )
+    x0 = x0 - inset.left;
+    y0 = y0 - inset.bottom;
+  endif
+
   if (numel(x0) > 1)
     x1 = max (x0) + width;
     y1 = max (y0) + height;
     x0 = min (x0);
     y0 = min (y0);
     pos = [x0, y0, x1-x0, y1-y0];
   else
     pos = [x0, y0, width, height];
